internal/contentmanager/contentmanager.go

```diff
@@ -2,222 +2,143 @@ package contentmanager
-	if err := v1alpha1.AddToScheme(scheme); err != nil {
-		return nil, fmt.Errorf("adding operator controller APIs to scheme: %w", err)
+// NewManager creates a new Manager
+func NewManager(rcm RestConfigMapper, cfg *rest.Config, mapper meta.RESTMapper, opts ...ManagerOption) Manager {
+	m := &managerImpl{
+		rcm:          rcm,
+		baseCfg:      cfg,
+		caches:       make(map[string]cmcache.Cache),
+		mapper:       mapper,
+		mu:           &sync.Mutex{},
+		syncTimeout:  time.Second * 10,
+		resyncPeriod: time.Hour * 10,
 	}
 
-	for _, obj := range objs {
-		gvk := obj.GetObjectKind().GroupVersionKind()
-
-		// If the Kind or Version is not set in an object's GroupVersionKind
-		// attempting to add it to the runtime.Scheme will result in a panic.
-		// To avoid panics, we are doing the validation and returning early
-		// with an error if any objects are provided with a missing Kind or Version
-		// field
-		if gvk.Kind == "" {
-			return nil, fmt.Errorf(
-				"adding %s to scheme; object Kind is not defined",
-				obj.GetName(),
-			)
-		}
-
-		if gvk.Version == "" {
-			return nil, fmt.Errorf(
-				"adding %s to scheme; object Version is not defined",
-				obj.GetName(),
-			)
-		}
-
-		listKind := gvk.Kind + "List"
-
-		if !scheme.Recognizes(gvk) {
-			// Since we can't have a mapping to every possible Go type in existence
-			// based on the GVK we need to use the unstructured types for mapping
-			u := &unstructured.Unstructured{}
-			u.SetGroupVersionKind(gvk)
-			ul := &unstructured.UnstructuredList{}
-			ul.SetGroupVersionKind(gvk.GroupVersion().WithKind(listKind))
-
-			scheme.AddKnownTypeWithName(gvk, u)
-			scheme.AddKnownTypeWithName(gvk.GroupVersion().WithKind(listKind), ul)
-			// Adding the common meta schemas to the scheme for the GroupVersion
-			// is necessary to ensure the scheme is aware of the different operations
-			// that can be performed against the resources in this GroupVersion
-			metav1.AddToGroupVersion(scheme, gvk.GroupVersion())
-		}
+	for _, opt := range opts {
+		opt(m)
 	}
 
-	return scheme, nil
+	return m
 }
 
-// Watch configures a controller-runtime cache.Cache and establishes watches for the provided resources.
-// It utilizes the provided ClusterExtension to set a DefaultLabelSelector on the cache.Cache
-// to ensure it is only caching and reacting to content that belongs to the ClusterExtension.
-// For each client.Object provided, a new source.Kind is created and used in a call to the Watch() method
-// of the provided controller.Controller to establish new watches for the managed resources.
-func (i *instance) Watch(ctx context.Context, ctrl controller.Controller, ce *v1alpha1.ClusterExtension, objs []client.Object) error {
-	if len(objs) == 0 || ce == nil || ctrl == nil {
-		return nil
+// Get returns the Cache for the provided ClusterExtension.
+// An error will be returned if:
+//   - The provided ClusterExtension is nil
+//   - There is no cache found for the provided ClusterExtension
```

Comment:
Update the comment here to reflect current state of the world

Reply:
addressed in https://github.com/operator-framework/operator-controller/pull/1119/commits/af075c5137065a5e0b2e0cfc1b4debbb367f792f